nemo/collections/nlp/models/text_classification_model.py

```diff
@@ -12,24 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dataclasses import dataclass
 from typing import Dict, Optional
 
+import hydra
 import torch
+from omegaconf import MISSING, DictConfig
 from torch.utils.data import DataLoader
 
 from nemo.collections.common.losses import CrossEntropyLoss
 from nemo.collections.common.tokenizers.bert_tokenizer import NemoBertTokenizer
 from nemo.collections.nlp.data.text_classification import TextClassificationDataDesc, TextClassificationDataset
 from nemo.collections.nlp.modules.common import SequenceClassifier
 from nemo.collections.nlp.modules.common.common_utils import get_pretrained_lm_model
-from nemo.core.classes import typecheck
-from nemo.core.classes.modelPT import ModelPT
+from nemo.core.classes.common import typecheck
+from nemo.core.classes.modelPT import ModelPT, ModelPTConfig
 from nemo.core.neural_types import NeuralType
 from nemo.utils.decorators import experimental
 
 __all__ = ['TextClassificationModel']
 
 
+@dataclass
+class TextClassificationDataConfig:
```

Comment:
While not a strict requirement for phase 2 (hybrid development with config objects and yaml), it's good practice and helps with auto complete.

Reply:
These classes will ultimately give us a common look and feel across nemo and nemo collections because they are hierarchical.

Additionally, we can use these classses for structured configs: https://hydra.cc/docs/next/tutorials/structured_config/intro which will allow us to validate configuration files.

Reply:
I wouldn't require to configure everything with Config objects. Let's just require it for models.
On the other hand, if the author choses to configure their classes other than models with configs - that should be OK too.

Reply:
IMHO, adding 5 or 6 classes just for supporting input arguments of a simple text classification model looks too much. Something which can be done in 20 lines of Pytorch and Python. But if there is no way around having them, we can keep them.